Makefile

```diff
@@ -12,153 +12,82 @@
-# integration_test_image models the integration-test-image resource in CI.
-
-# Fetch the image.
-.PHONY: integration_test_image
-integration_test_image:
-	docker pull ${DOCKER_IMAGE_INTEGRATION_URI}
-## END Resources
+		-e SERVICE_ACCOUNT_JSON \
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/usr/local/bin/execute_with_credentials.sh prepare_environment
+
+# Clean up test environment within the docker container
+.PHONY: docker_test_cleanup
+docker_test_cleanup:
+	docker run --rm -it \
+		-e SERVICE_ACCOUNT_JSON \
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/usr/local/bin/execute_with_credentials.sh cleanup_environment
+
+# Execute integration tests within the docker container
+.PHONY: docker_test_integration
+docker_test_integration:
+	docker run --rm -it \
+		-e SERVICE_ACCOUNT_JSON \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/usr/local/bin/test_integration.sh
 
-# Entry point from inside the container executing the test suite.  Modeled
-# after project factory Makefile structure and behavior.  The pipeline should
-# call this make target in a run step.
-.PHONY: test_integration
-test_integration:
-	./test/ci_integration.sh
+# Execute lint tests within the docker container
+.PHONY: docker_test_lint
+docker_test_lint:
+	docker run --rm -it \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/usr/local/bin/test_lint.sh
 
-## BEGIN Job Tasks
-# Model the CI integration-tests job definition and tasks as a make target.
-# The Git repo is placed in the same path the CI pull-request resource will be
-# placed inside the image.  Params are passed in the same way params are passed
-# to the container inside of CI.
-#
-# Unlike other modules which use a *.tfvars file, this module uses environment
-# variables to pass inputs into the integration tests.  Pass along the same
-# variables set as parameters of the CI Job.
-.PHONY: test_integration_docker
-test_integration_docker: integration_test_image
+# Execute bats tests within the docker container
+.PHONY: docker_test_bats
+docker_test_bats:
 	docker run --rm -it \
-		--volume $(CURDIR):/terraform-google-startup-scripts \
-		--workdir /terraform-google-startup-scripts \
-		--env SERVICE_ACCOUNT_JSON \
-		--env PROJECT_ID \
-		--env REGION \
-		${DOCKER_IMAGE_INTEGRATION_URI} \
-		/bin/bash -c make test_integration
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/bin/bash -c 'time find test/spec/ -name *.bats -print0 | xargs -0 -P4 --no-run-if-empty -n1 bats'
```

Comment:
We might want to consider putting this command into a helper function within a custom task_helper_functions file [like this](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/task_helper_functions.sh#L24)) so it can be references from both make and cloud shell.

Reply:
Very good point, I missed this piece, it's done.